apps/browser-extension-wallet/src/hooks/useWalletManager.ts

```diff
@@ -528,6 +574,78 @@ export const useWalletManager = (): UseWalletManager => {
     [getCurrentChainId]
   );
 
+  const createSharedWallet = async ({
+    mnemonic,
+    name,
+    password,
+    chainId = getCurrentChainId(),
+    publicKeys,
+    ownSignerWalletId
+  }: CreateSharedWalletParams): Promise<Wallet.CardanoWallet> => {
+    const accountIndex = 0;
+    const passphrase = Buffer.from(password, 'utf8');
+    const keyAgent = await Wallet.KeyManagement.InMemoryKeyAgent.fromBip39MnemonicWords(
+      {
+        chainId,
+        getPassphrase: async () => passphrase,
+        mnemonicWords: mnemonic,
+        accountIndex,
+        purpose: KeyManagement.KeyPurpose.MULTI_SIG
+      },
+      {
+        bip32Ed25519: Wallet.bip32Ed25519,
+        logger
+      }
+    );
+
+    const multiSigParticipants = publicKeys.sort();
+
+    const paymentScript = await buildScript(multiSigParticipants, KeyManagement.KeyRole.External);
+    const stakingScript = await buildScript(multiSigParticipants, KeyManagement.KeyRole.Stake);
+
+    const createScriptWalletProps = {
+      accountIndex: 0,
+      metadata: { name },
+      ownSigners: [
+        {
+          accountIndex: 0,
+          paymentScriptKeyPath: {
+            index: 0,
+            role: KeyManagement.KeyRole.External
+          },
```

Comment:
Would be better to have a variable for this, and also pass full path to `buildScript` so that if/when we want to use a different index, it only has to be modified in a single place. Same for staking key path.
```suggestion
          paymentScriptKeyPath,
```